From a6d3fa93eef58cbfb3baed61034768de4d129d1e Mon Sep 17 00:00:00 2001
From: Rahul Akolkar <rahul@apache.org>
Date: Sat, 12 Jul 2008 03:15:45 +0000
Subject: [PATCH] Porting r676121 from trunk. May 08 WD change: <data> uses
 "name" attribute instead of "ID" Thanks to Edzard Hoefig <info at edzard dot
 net> for the patch. SCXML-74 Also deprecated get/setName methods and added
 Edzard to contributors list.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/scxml/branches/J6@676122 13f79535-47bb-0310-9956-ffa450edef68
---
 pom.xml                                       |  3 ++
 project.xml                                   |  3 ++
 .../org/apache/commons/scxml/SCXMLHelper.java |  6 ++--
 .../commons/scxml/io/SCXMLSerializer.java     |  4 +--
 .../org/apache/commons/scxml/model/Data.java  | 32 ++++++++++++++++---
 .../scxml/custom-hello-world-04-el.xml        |  2 +-
 .../scxml/custom-hello-world-04-jexl.xml      |  2 +-
 .../scxml/env/javascript/JSEvaluatorTest.java |  2 +-
 .../scxml/env/javascript/example-01.xml       |  2 +-
 .../commons/scxml/env/jexl/datamodel-01.xml   |  6 ++--
 .../commons/scxml/env/jexl/datamodel-02.xml   |  6 ++--
 .../commons/scxml/env/jexl/datamodel-03.xml   | 22 ++++++-------
 .../commons/scxml/env/jexl/datamodel-04.xml   |  2 +-
 .../commons/scxml/env/jexl/eventdata-03.xml   |  2 +-
 .../commons/scxml/env/jsp/datamodel-01.xml    |  6 ++--
 .../commons/scxml/env/jsp/datamodel-02.xml    |  6 ++--
 .../commons/scxml/env/jsp/datamodel-03.xml    | 22 ++++++-------
 .../commons/scxml/env/xpath/example-01.xml    |  2 +-
 .../commons/scxml/issues/issue64-01.xml       |  2 +-
 .../commons/scxml/issues/issue64-02.xml       |  4 +--
 .../commons/scxml/model/assign-test.xml       |  8 ++---
 21 files changed, 86 insertions(+), 58 deletions(-)

diff --git a/pom.xml b/pom.xml
index 59f32575e..3b3e0abac 100644
--- a/pom.xml
+++ b/pom.xml
@@ -108,6 +108,9 @@
     <contributor>
       <name>Elaine Wong</name>
     </contributor>
+    <contributor>
+      <name>Edzard Hoefig</name>
+    </contributor>
   </contributors>
 
   <dependencies>
diff --git a/project.xml b/project.xml
index 4dc860c5e..759861db7 100644
--- a/project.xml
+++ b/project.xml
@@ -155,6 +155,9 @@
     <contributor>
       <name>Elaine Wong</name>
     </contributor>
+    <contributor>
+      <name>Edzard Hoefig</name>
+    </contributor>
   </contributors>
   
   <dependencies>
diff --git a/src/main/java/org/apache/commons/scxml/SCXMLHelper.java b/src/main/java/org/apache/commons/scxml/SCXMLHelper.java
index 618aa49a0..8a11cc672 100644
--- a/src/main/java/org/apache/commons/scxml/SCXMLHelper.java
+++ b/src/main/java/org/apache/commons/scxml/SCXMLHelper.java
@@ -435,7 +435,7 @@ public static void cloneDatamodel(final Datamodel datamodel,
             }
             // prefer "src" over "expr" over "inline"
             if (!SCXMLHelper.isStringEmpty(datum.getSrc())) {
-                ctx.setLocal(datum.getName(), valueNode);
+                ctx.setLocal(datum.getId(), valueNode);
             } else if (!SCXMLHelper.isStringEmpty(datum.
                     getExpr())) {
                 Object value = null;
@@ -451,9 +451,9 @@ public static void cloneDatamodel(final Datamodel datamodel,
                         defaultLog.error(see.getMessage(), see);
                     }
                 }
-                ctx.setLocal(datum.getName(), value);
+                ctx.setLocal(datum.getId(), value);
             } else {
-                ctx.setLocal(datum.getName(), valueNode);
+                ctx.setLocal(datum.getId(), valueNode);
             }
         }
     }
diff --git a/src/main/java/org/apache/commons/scxml/io/SCXMLSerializer.java b/src/main/java/org/apache/commons/scxml/io/SCXMLSerializer.java
index 1a41d4c5c..7087cbedb 100644
--- a/src/main/java/org/apache/commons/scxml/io/SCXMLSerializer.java
+++ b/src/main/java/org/apache/commons/scxml/io/SCXMLSerializer.java
@@ -350,8 +350,8 @@ public static void serializeDatamodel(final StringBuffer b,
                     }
                     b.append(indent).append(INDENT).append(out.toString());
                 } else {
-                    b.append(indent).append(INDENT).append("<data name=\"").
-                        append(datum.getName()).append("\" expr=\"").
+                    b.append(indent).append(INDENT).append("<data id=\"").
+                        append(datum.getId()).append("\" expr=\"").
                         append(datum.getExpr()).append("\" />\n");
                 }
             }
diff --git a/src/main/java/org/apache/commons/scxml/model/Data.java b/src/main/java/org/apache/commons/scxml/model/Data.java
index acf481902..3e268c645 100644
--- a/src/main/java/org/apache/commons/scxml/model/Data.java
+++ b/src/main/java/org/apache/commons/scxml/model/Data.java
@@ -34,9 +34,10 @@ public class Data implements NamespacePrefixesHolder, Serializable {
     private static final long serialVersionUID = 1L;
 
     /**
-     * The name of this data instance, that is used as its identifier.
+     * The identifier of this data instance.
+     * For backwards compatibility this is also the name.
      */
-    private String name;
+    private String id;
 
     /**
      * The URL to get the XML data tree from.
@@ -65,7 +66,7 @@ public class Data implements NamespacePrefixesHolder, Serializable {
      * Constructor.
      */
     public Data() {
-        this.name = null;
+        this.id = null;
         this.src = null;
         this.expr = null;
         this.node = null;
@@ -75,18 +76,39 @@ public Data() {
      * Get the name.
      *
      * @return String The name.
+     * @deprecated Use {@link #getId()} instead.
      */
     public final String getName() {
-        return name;
+        return id;
     }
 
     /**
      * Set the name.
      *
      * @param name The name.
+     * @deprecated Use {@link #setId(String)} instead.
      */
     public final void setName(final String name) {
-        this.name = name;
+    	System.err.println("\n\n" + name + "\n\n"); 
+        this.id = name;
+    }
+
+    /**
+     * Get the id.
+     *
+     * @return String An identifier.
+     */
+    public final String getId() {
+        return id;
+    }
+
+    /**
+     * Set the id.
+     *
+     * @param id The identifier.
+     */
+    public final void setId(final String id) {
+        this.id = id;
     }
 
     /**
diff --git a/src/test/java/org/apache/commons/scxml/custom-hello-world-04-el.xml b/src/test/java/org/apache/commons/scxml/custom-hello-world-04-el.xml
index a76b154ff..57b642ce4 100644
--- a/src/test/java/org/apache/commons/scxml/custom-hello-world-04-el.xml
+++ b/src/test/java/org/apache/commons/scxml/custom-hello-world-04-el.xml
@@ -20,7 +20,7 @@
        xmlns:my="http://my.custom-actions.domain/CUSTOM"
        version="1.0" initialstate="init">
     <datamodel>
-        <data name="helloName1" />
+        <data id="helloName1" />
     </datamodel>
 
     <state id="init">        
diff --git a/src/test/java/org/apache/commons/scxml/custom-hello-world-04-jexl.xml b/src/test/java/org/apache/commons/scxml/custom-hello-world-04-jexl.xml
index a2e79306e..624e2a4b7 100644
--- a/src/test/java/org/apache/commons/scxml/custom-hello-world-04-jexl.xml
+++ b/src/test/java/org/apache/commons/scxml/custom-hello-world-04-jexl.xml
@@ -20,7 +20,7 @@
        xmlns:my="http://my.custom-actions.domain/CUSTOM"
        version="1.0" initialstate="init">
     <datamodel>
-        <data name="helloName1" />
+        <data id="helloName1" />
     </datamodel>
 
     <state id="init">        
diff --git a/src/test/java/org/apache/commons/scxml/env/javascript/JSEvaluatorTest.java b/src/test/java/org/apache/commons/scxml/env/javascript/JSEvaluatorTest.java
index 374fc5861..561b39ccf 100644
--- a/src/test/java/org/apache/commons/scxml/env/javascript/JSEvaluatorTest.java
+++ b/src/test/java/org/apache/commons/scxml/env/javascript/JSEvaluatorTest.java
@@ -61,7 +61,7 @@ public class JSEvaluatorTest extends TestCase {
                                                         "initialstate = 'start' "  +
                                                         "version      = '1.0'>" +
                                                   "<datamodel>"           +
-                                                  "<data name='forest'>"  +
+                                                  "<data id='forest'>"  +
                                                    "<tree xmlns=''>"      +
                                                    "<branch>"             +
                                                    "<twig>leaf</twig>"    +
diff --git a/src/test/java/org/apache/commons/scxml/env/javascript/example-01.xml b/src/test/java/org/apache/commons/scxml/env/javascript/example-01.xml
index c90f75a4a..fcac09c8a 100644
--- a/src/test/java/org/apache/commons/scxml/env/javascript/example-01.xml
+++ b/src/test/java/org/apache/commons/scxml/env/javascript/example-01.xml
@@ -24,7 +24,7 @@
  <!-- DATA MODEL -->
 
  <datamodel>
-  <data name='jungle'>
+  <data id='jungle'>
    <animals xmlns=''>
     <lion>
      <name>Simba</name>
diff --git a/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-01.xml b/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-01.xml
index 6da64d805..5bb3fdf18 100644
--- a/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-01.xml
+++ b/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-01.xml
@@ -24,7 +24,7 @@
 
     <!-- Root or document datamodel -->
     <datamodel>
-        <data name="docdata">
+        <data id="docdata">
             <root xmlns="">
                 <foo>
                     <bar>alpha</bar>
@@ -42,9 +42,9 @@
         <!-- datamodel scoped to state "main" -->
         <datamodel>
             <!-- Degenerate usage, similar to the <var> element -->
-            <data name="mainvar" expr="0" />
+            <data id="mainvar" expr="0" />
             <!-- Usage where the value is an XML data tree -->
-            <data name="maindata">
+            <data id="maindata">
                 <root xmlns="">
                     <a>
                         <b>
diff --git a/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-02.xml b/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-02.xml
index aaadde995..7278fa6b9 100644
--- a/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-02.xml
+++ b/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-02.xml
@@ -25,7 +25,7 @@
     <state id="ten">
 
         <datamodel>
-            <data name="data10" xmlns:ns1="http://namespace.test.domain/1">
+            <data id="data10" xmlns:ns1="http://namespace.test.domain/1">
                 <root xmlns="">
                     <ns1:foo>
                       <bar>10</bar>
@@ -48,7 +48,7 @@
 
         <datamodel>
             <!-- Start with a prefixless XPath -->
-            <data name="data20">
+            <data id="data20">
                 <ns1:root>
                     <ns2:foo>20</ns2:foo>
                 </ns1:root>
@@ -65,7 +65,7 @@
 
         <datamodel>
             <!-- Start with a prefixless XPath -->
-            <data name="data30">
+            <data id="data30">
                 <root xmlns="http://namespace.test.domain/1">
                     <foo xmlns="http://namespace.test.domain/2"
                          xmlns:ns1="http://namespace.test.domain/3"
diff --git a/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-03.xml b/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-03.xml
index c5ed31c6b..274d72394 100644
--- a/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-03.xml
+++ b/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-03.xml
@@ -29,7 +29,7 @@
      <datamodel>
 
         <!-- We'll use this for XPaths -->
-        <data name="rootdata"
+        <data id="rootdata"
               xmlns:ns1="scheme://namespace.test.domain/1"
               xmlns:ns2="scheme://namespace.test.domain/2"
               xmlns:ns3="scheme://namespace.test.domain/3">
@@ -42,7 +42,7 @@
 
         <!-- We'll use this for the JUnit test
              NamespacePrefixedPathsTest.java (scxml package) -->
-        <data name="retval" />
+        <data id="retval" />
 
     </datamodel>
 
@@ -51,7 +51,7 @@
 
         <datamodel>
 
-            <data name="data10" xmlns:ns4="scheme://namespace.test.domain/1">
+            <data id="data10" xmlns:ns4="scheme://namespace.test.domain/1">
                 <root xmlns="">
                     <ns4:foo>
                       <bar>10</bar>
@@ -87,7 +87,7 @@
 
         <datamodel>
 
-            <data name="data20">
+            <data id="data20">
                 <ns1:root>
                     <ns2:foo>20</ns2:foo>
                 </ns1:root>
@@ -136,7 +136,7 @@
 
         <datamodel>
 
-            <data name="data30">
+            <data id="data30">
                 <root xmlns="scheme://namespace.test.domain/1">
                     <foo xmlns="scheme://namespace.test.domain/2"
                          xmlns:ns1="scheme://namespace.test.domain/3"
@@ -163,13 +163,13 @@
 
         <datamodel>
 
-            <data name="data40">
+            <data id="data40">
                 <root xmlns="">
                     <ns1:foo ns2:attfoo="40"/>
                 </root>
             </data>
 
-            <data name="data41">
+            <data id="data41">
                 <ns3:root>
                     <ns4:foo>41</ns4:foo>
                 </ns3:root>
@@ -188,7 +188,7 @@
 
         <datamodel>
 
-            <data name="data50"  xmlns:ns1="scheme://namespace.test.domain/1"
+            <data id="data50"  xmlns:ns1="scheme://namespace.test.domain/1"
                                  xmlns:ns2="scheme://namespace.test.domain/2"
                                  xmlns:ns3="scheme://namespace.test.domain/3">
                 <ns1:root>
@@ -196,7 +196,7 @@
                 </ns1:root>
             </data>
 
-            <data name="data51" xmlns:ns3="scheme://namespace.test.domain/3"
+            <data id="data51" xmlns:ns3="scheme://namespace.test.domain/3"
                                 xmlns:ns4="scheme://namespace.test.domain/4">
                 <ns3:root>
                     <ns4:foo attfoo="51"/>
@@ -223,13 +223,13 @@
                    xmlns:ns3="scheme://namespace.test.domain/3"
                    xmlns:ns4="scheme://namespace.test.domain/4">
 
-            <data name="data60">
+            <data id="data60">
                 <root xmlns="">
                     <ns1:foo ns2:attfoo="60"/>
                 </root>
             </data>
 
-            <data name="data61">
+            <data id="data61">
                 <ns3:root>
                     <ns4:foo attfoo="61"/>
                 </ns3:root>
diff --git a/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-04.xml b/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-04.xml
index 916752312..c6392821d 100644
--- a/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-04.xml
+++ b/src/test/java/org/apache/commons/scxml/env/jexl/datamodel-04.xml
@@ -24,7 +24,7 @@
 
     <!-- Root or document datamodel -->
     <datamodel>
-        <data name="payload"/>
+        <data id="payload"/>
     </datamodel>
 
     <state id="ten">
diff --git a/src/test/java/org/apache/commons/scxml/env/jexl/eventdata-03.xml b/src/test/java/org/apache/commons/scxml/env/jexl/eventdata-03.xml
index a48d1e82b..ad1e1232c 100644
--- a/src/test/java/org/apache/commons/scxml/env/jexl/eventdata-03.xml
+++ b/src/test/java/org/apache/commons/scxml/env/jexl/eventdata-03.xml
@@ -21,7 +21,7 @@
        initialstate="ten">
 
     <datamodel>
-        <data name="rootdata">
+        <data id="rootdata">
             <root xmlns="">
                 <one>1</one>
                 <two>2</two>
diff --git a/src/test/java/org/apache/commons/scxml/env/jsp/datamodel-01.xml b/src/test/java/org/apache/commons/scxml/env/jsp/datamodel-01.xml
index 7a8391a06..7e4194fc9 100644
--- a/src/test/java/org/apache/commons/scxml/env/jsp/datamodel-01.xml
+++ b/src/test/java/org/apache/commons/scxml/env/jsp/datamodel-01.xml
@@ -24,7 +24,7 @@
 
     <!-- Root or document datamodel -->
     <datamodel>
-        <data name="docdata">
+        <data id="docdata">
             <root xmlns="">
                 <foo>foo</foo>
             </root>
@@ -39,8 +39,8 @@
 
         <!-- datamodel scoped to state "main" -->
         <datamodel>
-            <data name="mainvar" expr="${0}" />
-            <data name="maindata">
+            <data id="mainvar" expr="${0}" />
+            <data id="maindata">
                 <root xmlns="">
                     <foo>
                         <bar>bar</bar>
diff --git a/src/test/java/org/apache/commons/scxml/env/jsp/datamodel-02.xml b/src/test/java/org/apache/commons/scxml/env/jsp/datamodel-02.xml
index 2e6e45353..55b925e99 100644
--- a/src/test/java/org/apache/commons/scxml/env/jsp/datamodel-02.xml
+++ b/src/test/java/org/apache/commons/scxml/env/jsp/datamodel-02.xml
@@ -25,7 +25,7 @@
     <state id="ten">
 
         <datamodel>
-            <data name="data10">
+            <data id="data10">
                 <root xmlns="">
                     <foo>10</foo>
                 </root>
@@ -43,7 +43,7 @@
 
         <datamodel>
             <!-- Start with a prefixless XPath -->
-            <data name="data20">
+            <data id="data20">
                 <ns1:root>
                     <ns2:foo>20</ns2:foo>
                 </ns1:root>
@@ -60,7 +60,7 @@
 
         <datamodel>
             <!-- Start with a prefixless XPath -->
-            <data name="data30">
+            <data id="data30">
                 <root xmlns="http://namespace.test.domain/1">
                     <foo xmlns="http://namespace.test.domain/2">30</foo>
                 </root>
diff --git a/src/test/java/org/apache/commons/scxml/env/jsp/datamodel-03.xml b/src/test/java/org/apache/commons/scxml/env/jsp/datamodel-03.xml
index 8516637a4..0d0cc9235 100644
--- a/src/test/java/org/apache/commons/scxml/env/jsp/datamodel-03.xml
+++ b/src/test/java/org/apache/commons/scxml/env/jsp/datamodel-03.xml
@@ -29,7 +29,7 @@
      <datamodel>
 
         <!-- We'll use this for XPaths -->
-        <data name="rootdata"
+        <data id="rootdata"
               xmlns:ns1="scheme://namespace.test.domain/1"
               xmlns:ns2="scheme://namespace.test.domain/2"
               xmlns:ns3="scheme://namespace.test.domain/3">
@@ -42,7 +42,7 @@
 
         <!-- We'll use this for the JUnit test
              NamespacePrefixedPathsTest.java (scxml package) -->
-        <data name="retval" />
+        <data id="retval" />
 
     </datamodel>
 
@@ -51,7 +51,7 @@
 
         <datamodel>
 
-            <data name="data10" xmlns:ns4="scheme://namespace.test.domain/1">
+            <data id="data10" xmlns:ns4="scheme://namespace.test.domain/1">
                 <root xmlns="">
                     <ns4:foo>
                       <bar>10</bar>
@@ -87,7 +87,7 @@
 
         <datamodel>
 
-            <data name="data20">
+            <data id="data20">
                 <ns1:root>
                     <ns2:foo>20</ns2:foo>
                 </ns1:root>
@@ -136,7 +136,7 @@
 
         <datamodel>
 
-            <data name="data30">
+            <data id="data30">
                 <root xmlns="scheme://namespace.test.domain/1">
                     <foo xmlns="scheme://namespace.test.domain/2"
                          xmlns:ns1="scheme://namespace.test.domain/3"
@@ -163,13 +163,13 @@
 
         <datamodel>
 
-            <data name="data40">
+            <data id="data40">
                 <root xmlns="">
                     <ns1:foo ns2:attfoo="40"/>
                 </root>
             </data>
 
-            <data name="data41">
+            <data id="data41">
                 <ns3:root>
                     <ns4:foo>41</ns4:foo>
                 </ns3:root>
@@ -188,7 +188,7 @@
 
         <datamodel>
 
-            <data name="data50"  xmlns:ns1="scheme://namespace.test.domain/1"
+            <data id="data50"  xmlns:ns1="scheme://namespace.test.domain/1"
                                  xmlns:ns2="scheme://namespace.test.domain/2"
                                  xmlns:ns3="scheme://namespace.test.domain/3">
                 <ns1:root>
@@ -196,7 +196,7 @@
                 </ns1:root>
             </data>
 
-            <data name="data51" xmlns:ns3="scheme://namespace.test.domain/3"
+            <data id="data51" xmlns:ns3="scheme://namespace.test.domain/3"
                                 xmlns:ns4="scheme://namespace.test.domain/4">
                 <ns3:root>
                     <ns4:foo attfoo="51"/>
@@ -223,13 +223,13 @@
                    xmlns:ns3="scheme://namespace.test.domain/3"
                    xmlns:ns4="scheme://namespace.test.domain/4">
 
-            <data name="data60">
+            <data id="data60">
                 <root xmlns="">
                     <ns1:foo ns2:attfoo="60"/>
                 </root>
             </data>
 
-            <data name="data61">
+            <data id="data61">
                 <ns3:root>
                     <ns4:foo attfoo="61"/>
                 </ns3:root>
diff --git a/src/test/java/org/apache/commons/scxml/env/xpath/example-01.xml b/src/test/java/org/apache/commons/scxml/env/xpath/example-01.xml
index f7293d502..2bf8c5536 100644
--- a/src/test/java/org/apache/commons/scxml/env/xpath/example-01.xml
+++ b/src/test/java/org/apache/commons/scxml/env/xpath/example-01.xml
@@ -21,7 +21,7 @@
        version="1.0">
 
     <datamodel>
-        <data name="jungle">
+        <data id="jungle">
             <animals xmlns="">
                 <lion>
                     <name>Simba</name>
diff --git a/src/test/java/org/apache/commons/scxml/issues/issue64-01.xml b/src/test/java/org/apache/commons/scxml/issues/issue64-01.xml
index 018a126a9..95d4d9ef3 100644
--- a/src/test/java/org/apache/commons/scxml/issues/issue64-01.xml
+++ b/src/test/java/org/apache/commons/scxml/issues/issue64-01.xml
@@ -22,7 +22,7 @@
 
     <state id="tranbug">
          <datamodel>
- 			<data name="dummy" expr="'somedata'"/>
+ 			<data id="dummy" expr="'somedata'"/>
          </datamodel>
         <onentry>
             <log expr="'Begin transition bug test ...'" />
diff --git a/src/test/java/org/apache/commons/scxml/issues/issue64-02.xml b/src/test/java/org/apache/commons/scxml/issues/issue64-02.xml
index c3dce5fbe..bccffb930 100644
--- a/src/test/java/org/apache/commons/scxml/issues/issue64-02.xml
+++ b/src/test/java/org/apache/commons/scxml/issues/issue64-02.xml
@@ -22,7 +22,7 @@
        initialstate="tranbug">
 
     <datamodel>
-        <data name="foo">
+        <data id="foo">
             <alpha xmlns="">
                 <beta/>
             </alpha>
@@ -39,7 +39,7 @@
         <transition event="show.bug" target="end">
          <!-- For example, FOLLOWING datamodel IS MISPLACED -->
          <datamodel>
- 			<data name="dummy" expr="'somedata'"/>
+ 			<data id="dummy" expr="'somedata'"/>
          </datamodel>
          <log expr="dummy" />
          <log expr="'*****' + dummy" />
diff --git a/src/test/java/org/apache/commons/scxml/model/assign-test.xml b/src/test/java/org/apache/commons/scxml/model/assign-test.xml
index 2d1f983c9..ed7e97706 100644
--- a/src/test/java/org/apache/commons/scxml/model/assign-test.xml
+++ b/src/test/java/org/apache/commons/scxml/model/assign-test.xml
@@ -24,12 +24,12 @@
   <state id="assign1" final="true">
 
     <datamodel>
-        <data name="foo">
+        <data id="foo">
             <root xmlns="">
                 <foo/>
             </root>
         </data>
-        <data name="bar">
+        <data id="bar">
             <root xmlns="">
                 <bar>5</bar>
             </root>
@@ -48,10 +48,10 @@
   <state id="assign2">
 
     <datamodel>
-      <data name="jira51data1">
+      <data id="jira51data1">
           <rad:timeout>10</rad:timeout>
       </data> 
-      <data name="jira51data2">
+      <data id="jira51data2">
           <rad:short xmlns="">20</rad:short>
       </data>
     </datamodel>
